File: datascience/lab1/homework/q5.R
raw <- readline("Enter a number (default 5): ")
n <- if (nchar(raw) == 0) 5 else as.integer(raw)

if (n < 0) {
  cat("Factorial is not defined for negative numbers.\n")
} else {
  factorial_value <- 1
  for (i in seq_len(n)) {
    factorial_value <- factorial_value * i
  }

  cat(sprintf("%d! = %g\n", n, factorial_value))
}

File: datascience/lab1/py/homework/q5.py
n = int(input("Enter a number (default 5): ").strip() or 5)

if n < 0:
    print("Factorial is not defined for negative numbers.")
else:
    factorial = 1
    for i in range(1, n + 1):
        factorial *= i

    print(f"{n}! = {factorial}")
